unyt/dimensions.py

```diff
@@ -290,14 +290,14 @@ def new_f(*args, **kwargs):
     return check_accepts
 
 
-def returns(r_unit):
+def returns(*r_unit):
```

Comment:
this is a breaking API change to any user code that would use the argument as keyword
```python
@returns(r_unit=time)
def foo():
    ...
```
It is however possible to keep it backward compatible for now and deprecate this cumbersome and undocumented use case, like so:
```suggestion
def returns(*r_units, r_unit):
```
and with internal modifications to the function to handle all special cases where `r_unit` is provide:
  - raise an error if `r_units` is not empty
  - raise a warning that passing `r_unit` as keyword is deprecated, and assign`r_units = (r_unit,)`

Reply:
Good catch! Any better now?

Reply:
yes, you're doing great ! And well done on the default value for the deprecated argument, I see now that I forgot to mention it.